lib/makara/proxy.rb

```diff
@@ -76,6 +76,19 @@ def stick_to_master!(persist = true)
       Makara::Context.stick(@id, stickiness_duration)
     end
 
+    def with_master
+      stick_to_master!
+      yield if block_given?
+    ensure
+      release_master!
```

Comment:
there's something here about if it was stuck to master before you ran this block.
Imagine if you were already stuck to master in a request, then called `with_master`  -  at the end of the block, I would think you'd recover the previous context (stuck to master).
I can imagine this is why there was an instance variable used in `without_sticking` and then leveraged in the various methods.

Reply:
good point, ill update.